### DISCUSSION TITLE
Missing Parts&Materials on AR mode

### DISCUSSION THREAD

**Discussion by sercandemirtas** [google/model-viewer#2211] [Q&A]
Hello all.

As you see in the picture on AR mode some parts are missing on my model. I created this "heater" on Blender 2.82 
I checked double-side (backface culling) material option and it seems there is no problem. Also all materials options are same for backface culling. 

What is the problem ?

![IMG_0471](https://user-images.githubusercontent.com/68244546/88983603-27272d00-d2d4-11ea-90ab-7772d44d9195.jpg)

**Comment by elalish** [comment]
From your images it appears the rendering is correct in model-viewer and incorrect in the SceneViewer AR mode? Have you tried `ar-modes="webxr"`? That should give consistent rendering. If you can remix our Glitch from the new issue instructions, I can pass the bug along to the SceneViewer team.

**Comment by sercandemirtas** [comment]
Im sorry but I dont understand why do I have to try ar-modes="webxr". I checked https://modelviewer.dev/ about webxr and  it looks like different viewer option with multiple models. My model looks normal in web modes. But when we click  AR button it crashes.

![IMG_0482](https://user-images.githubusercontent.com/68244546/89028404-f1fef700-d334-11ea-8f98-075af5df5976.PNG)

**Comment by sercandemirtas** [comment]
In the picture it looks like low resolution but it is not. The problem is missing parts&materials on model.

**Comment by elalish** [comment]
@sercandemirtas The webXR example simply shows how you *can* do fancy things like multiple models, but that's all just extra HTML. On its own, it's very much like SceneViewer: look also at https://modelviewer.dev/examples/augmented-reality.html

Now you're reporting a crash, it sounds like? Can I have a detailed repro for that and any console logs?

**Comment by sercandemirtas** [comment]
> @sercandemirtas The webXR example simply shows how you _can_ do fancy things like multiple models, but that's all just extra HTML. On its own, it's very much like SceneViewer: look also at https://modelviewer.dev/examples/augmented-reality.html
> 
> Now you're reporting a crash, it sounds like? Can I have a detailed repro for that and any console logs?

There is no crash. My bad. I mean "crash" with problem. "The problem is missing parts&materials on model." I said before that is the topic.

**Comment by elalish** [comment]
@sercandemirtas Okay, so in order for me to pass this bug to SceneViewer, I'm going to need your model. Can you put a link to it here? Also, have you tested that webXR mode does in fact render it properly?

**Comment by sercandemirtas** [comment]
I tried webXR. It rendered properly. I think there is problem while I converting glb to usdz. Im using Mac terminal usdzconvert command to convert it.  I upload my model glb + usdz format to wetransfer.  (https://we.tl/t-0coACcKFwq  4mb) It is not first time to convert usdz with mac terminal for me. I tried with others models there was no problem but I got this problem with "heater" model.

You can look preview of model (usdz) its look  weird. Not like original model.

<img width="216" alt="Ekran Resmi 2020-07-31 19 43 19" src="https://user-images.githubusercontent.com/68244546/89057325-1aebb000-d366-11ea-9988-19e995972a84.png">

**Comment by sercandemirtas** [comment]
And that is the link for my model if you wanna test. https://awesome-shockley-7e009f.netlify.app/responsive.html

**Comment by elalish** [comment]
Ah, you didn't mention you were on iOS. in this case it's a problem with QuickLook, not SceneViewer. You should post this question on Apple's forums. In future, please use the issue template instead of deleting it :)

**Comment by sercandemirtas** [comment]
Im sory bro. Im new on github now I understand why we have issue tepmlate. Thank you for trying to help me!

**Comment by sercandemirtas** [comment]
Problem fixed. It is all about solidify of object. If you are using blender or other 3d programs make your object solid. At least 0.001cm and do not forget to **_apply_** it. and also if you used modifier like "array" apply it too before export. Sometimes array objects could be problem when you convert to usdz..

**Comment by mlhw** [comment]
I'm also experiencing missing parts on my models but only in android devices. It was working fine when we created our site but has now stopped working.  It is still working on one of my test devices, running Android 10 (june security patch), Chrome 80.0.3987.99

**Works on:**
* Android 10 (june security patch), Chrome 80.0.3987.99

**Fails on/ can be reproduced on:**
* Android 10, Chrome 85
* OxygenOS  10.3.5, Chrome 85.0.4183.120
* Oxygen OS 10.0.9.HD01BA (Android 10)
* Android 11, Chrome 85.0.4183.127
* Android 11, Chrome version 85.0.4183.127

The devices are running latest version of Google play services for AR

The model renders correct on the web
![web](https://user-images.githubusercontent.com/13869815/94652101-ce870700-02f9-11eb-97a4-76cfa8444629.jpg)

But looses most parts in the AR mode
![concreteanswers husqvarnac](https://user-images.githubusercontent.com/13869815/94652180-e1014080-02f9-11eb-8f07-46ca418d67ff.jpg)

￼Everything is working fine on iOS devices
Models can be found on https://concreteanswers.husqvarnacp.com/

**Comment by elalish** [comment]
@mlhw Thanks, that looks like a pretty clear SceneViewer bug. I know we're tracking some problems around Android 11 with AR. Is Android 10 working in WebXR mode (our default) at least? @tpsiaki regarding the missing objects.

**Comment by tpsiaki** [comment]
Hi @mlhw - I can confirm this is a bug in Scene Viewer.  The specific issue is getting triggered here because your model has multiple nodes that share the same name.  We'll work on a fix, but in the meantime if you either remove the node names in your model, or rename the nodes to each have unique names, that should resolve this problem for you.

**Comment by mlhw** [comment]
Hi @tpsiaki - Thank you for the workaround. I can confirm that using unique node names or removing the none names completely resulted in a complete models again.
